### ISSUE TITLE
[Preview Card] component is inaccessible and has no warning in the docs

### ISSUE THREAD

**Issue by grace-snow** [mui/base-ui#4170]
# Bug report

## Current behavior

There is no way to activate the Preview Card popup using a keyboard, screen reader, touch or speech input. There is no warning against using it or its accessibility limitations in the docs. 

This fails [WCAG 2.2 Success Criterion 2.1.1 Keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html)

Additionally, the examples have been placed on links that fail [WCAG 2.2 Success Criterion 1.4.1 Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html) because they are links with blue text and no other style differentiation to show that they are links (e.g. an underline)

## Expected behavior

Preferably, remove this component. It isn't necessary, and creates an inequitable experience between users of different modalities.

If it cannot be removed:
- document that it is inaccessible to keyboard, screen reader, touch and speech input users and update Base Ui's accessibility statement 
- Or update the component to include a separate control to open the preview card popover

Ensure all links in the examples have an underline so it is visually clear they are links. 

## Reproducible example

Steps to reproduce: 
1. Navigate to the [Preview Card](https://base-ui.com/react/components/preview-card) component demo
2. Using a keyboard, focus on the "typography" link inside the component demo
3. Attempt to open the link preview using only a keyboard

## Base UI version

v1.2.0

## Which browser are you using?

Chrome 

## Which OS are you using?

Mac OS Tahoe 26.2

## Which assistive tech are you using (if applicable)?

Keyboard, VoiceOver, Apple Voice Control

## Additional context

The simplest solution would be to remove/retire this component from Base UI. 

Alternatively, a separate control could be added that renders after the links to reveal the preview card. This control would need an accessible name and the card would need to operate similar to a popover or tooltip, with all content announced to screen readers and persistence until it is dismissed.

**Comment by atomiks** [comment]
This is expected for Preview Card (similar to Tooltip):

> A popup that appears when a link is **hovered**, showing a preview for **sighted users**.

It doesn't work for touch or screen reader users as it's purely an enhancement for sighted keyboard/mouse pointer use, but this isn't an issue provided the Preview Card doesn't contain any content not accessible via the link itself (as the name implies and common use cases on the web show, e.g. Wikipedia's preview cards). There should a usage guideline section for this though.

**Comment by grace-snow** [comment]
I recommend improving the component demo to include underlined links as well. The component examples should be demonstrating accessible implementation.

**Comment by atomiks** [comment]
Yeah, the PR also updated the demo to show an underline (also consistent with our docs elsewhere, the demo is older than the docs-wide underline change to links we made)

<img width="503" height="274" alt="Image" src="https://github.com/user-attachments/assets/5bf5a42a-1f1a-4415-85ba-eb2cb54ee982" />

https://base-ui.com/react/components/preview-card
